npm/puppeteer/README.md

```diff
@@ -0,0 +1,360 @@
+
+    cy
+    .puppeteer('switchToTabAndGetContent')
+    .should('equal', 'You said: Hello from Page 1')
+  })
+```
+
+## API
+
+### Cypress Config - setup
+
+This sets up `@cypress/puppeteer` message handlers that run Puppeteer browser automation.
+
+```typescript
+setup(options)
+```
+
+#### Options
+
+- `on` _required_: The `on` event registration function provided by `setupNodeEvents`
+- `onMessage` _required_: An object with string keys and function values (see more details [below](#onmessage))
+- `puppeteer` _optional_: The `puppeteer` library imported from `puppeteer-core`, overriding the default version of `puppeteer-core` used by this plugin
+
+#### onMessage
+
+Functions provided in the `onMessage` option receive the following arguments:
+
+##### browser
+
+A [puppeteer browser instance](https://pptr.dev/api/puppeteer.browser) connected to the Cypress-launched browser.
+
+##### ...args
+
+The rest of the arguments are de-serialized arguments as passed to the `cy.puppeteer()` command from your Cypress test.
+
+### Cypress Config - retry
+
+This is a utility function provided to aid in retrying actions that may initially fail.
+
+```typescript
+retry(functionToRetry[, options])
+```
+
+#### functionToRetry
+
+_required_
+
+A function that will run and retry if an error is thrown. If an error is not thrown, `retry` will return the value returned by this function.
+
+The function will continue to run at the default or configured interval until the default or configured timeout, at which point `retry` will throw an error and cease retrying this function.
+
+#### Options
+
+_optional_
+
+- `timeout` _optional_: The total time in milliseconds during which to attempt retrying the function. Default: `4000ms`
+- `delayBetweenTries` _optional_: The time to wait between retries. Default: `200ms`
+
+### Cypress Spec - cy.puppeteer()
+
+```typescript
+cy.puppeteer(messageName[, ...args])
+```
+
+#### messageName
+
+_required_
+
+A string matching one of the keys passed to the `onMessage` option of `setup` in your Cypress config.
+
+#### ...args
+
+_optional_
+
+Values that will be passed to the message handler. These values must be JSON-serializable.
+
+Example:
+
+```typescript
+// spec
+cy.puppeteer('testNewTab', 'value 1', 42, [true, false])
+
+// Cypress config
+setup({
+  on,
+  onMessage: {
+    testNewTab (browser, stringArg, numberArg, arrayOfBooleans) {
+      // stringArg === 'value 1'
+      // numberArg === 42
+      // arrayOfBooleans[0] === true / arrayOfBooleans[1] === false
+    }
+  }
+})
+```
+
+## Examples
+
+These examples can be found and run in the [Cypress tests of this package](./cypress) with this project's [cypress.config.ts](./cypress.config.ts).
+
+While these example use tabs, they could just as easily apply to windows. Tabs and windows are essentially the same things as far as Puppeteer is concerned and encapsuled by instances of the [Page class](https://pptr.dev/api/puppeteer.page/).
```

Comment:
```suggestion
While these examples use tabs, they could just as easily apply to windows. Tabs and windows are essentially the same things as far as Puppeteer is concerned and encapsulated by instances of the [Page class](https://pptr.dev/api/puppeteer.page/).
```

Reply:
Addressed in https://github.com/cypress-io/cypress/pull/28370/commits/1d8356a92cf050de9c63813c72115bfcb160418e